src/core/DataFile.cpp

```diff
@@ -1960,5 +1960,64 @@ unsigned int DataFile::legacyFileVersion()
 	return std::distance( UPGRADE_VERSIONS.begin(), firstRequiredUpgrade );
 }
 
+// Change loops' filenames in <sampleclip>s
+void DataFile::upgrade_loopsRename()
+{
```

Comment:
Now this needs to be moved to after SampleAndHold routine. About here: https://github.com/LMMS/lmms/blob/de332820f7ed5ca33d64ffd62b8da147f6105c9a/src/core/DataFile.cpp#L1812